pkg/controller/delivery/reconciler.go

```diff
@@ -68,27 +69,59 @@ func (r *Reconciler) reconcileDelivery(ctx context.Context, delivery *v1alpha1.C
 	var resourcesNotFound []string
 
 	for _, resource := range delivery.Spec.Resources {
-		template, err := r.Repo.GetDeliveryTemplate(ctx, resource.TemplateRef)
-		if err != nil {
-			log.Error(err, "failed to get delivery cluster template", "template", resource.TemplateRef)
-			return controller.NewUnhandledError(fmt.Errorf("failed to get delivery cluster template: %w", err))
+		if resource.TemplateRef.Name != "" {
```

Comment:
I wonder if we could split these blocks into functions - I don't think there's anything wrong here but just looks a little too big to read 😅